cmd/temporalite/main.go

```diff
@@ -155,6 +157,11 @@ func buildCLI() *cli.App {
 					return cli.Exit(fmt.Sprintf("ERROR: only one of %q or %q flags may be passed at a time", ephemeralFlag, dbPathFlag), 1)
 				}
 
+				err := ensureDirExists(filepath.Dir(c.String(dbPathFlag)))
```

Comment:
I'm of mixed opinions about whether we should automatically create _any_ directory, or only do this when falling back to the default path in the user's home directory. See also https://github.com/temporalio/temporalite/issues/55#issuecomment-1118193801

Reply:
I'm not too concerned about it for this type of utility. I have multiple notes here:

* If you are concerned, you can default the DB to the current working dir, but sadly that'd be a compatibility change at this point
* If it would make you feel better, you can only lazily create the path when using the default
* If it's clearly logged the path of DB that is being used on program start, I think that's good enough
* Alternatively, we could just not have a default DB path at all

Reply:
I am not concerned about it in this case. For the default case, it makes sense. In the case the user specifies the path explicitly, I would expect it to exist in most cases. If not, the user gets, in the worst case, a directory created at an unexpected location which he can move or delete.

Reply:
> the user gets, in the worst case, a directory created at an unexpected location which he can move or delete.

I think the _worst_ case is that someone manages to mangle the flag in a production server, and Temporalite starts up without issue but now with a fresh database. It could be non-obvious at first since the server is healthy, but old data is missing. At that point you could have two separate databases with prod data that can't easily be merged.

> If it would make you feel better, you can only lazily create the path when using the default

This is what I was suggesting... but I'm probably being overly conservative.

Reply:
I'd support that. @hferentschik - think you can change to only `MkdirAll` if the user didn't specify a value explicitly?

Reply:
Yeah looking around, I think you might be hard pressed to find other databases that behave this way. Let's fix the getting started instructions and reconsider other cases if we get further user feedback.